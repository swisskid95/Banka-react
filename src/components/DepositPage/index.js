/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { Component } from 'react';
import SideNavPanel from '../commons/SideNavPanel';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import TopSection from '../commons/TopSection';
import PageTitle from '../commons/PageTitle';
import {
  fetchAccountDetails,
  performDeposit
} from '../../redux/actions/cashierStaffAction';

export class DepositPage extends Component {
  state = {
    accountNumber: '',
    remarks: '',
    amount: '',
    isLoading: false,
    showModal: false
  };

  handleInputChange = event => {
    event.preventDefault();
    const { target } = event;
    this.setState({ [target.name]: target.value });
  };

  handleConfirmDetails = async event => {
    const { fetchAccountDetails, user } = this.props;
    event.preventDefault();
    this.setState({ isLoading: true });
    fetchAccountDetails(user.token, this.state.accountNumber);
    this.setState({ isLoading: false });
    this.setState({ showModal: true });
  };

  handleModalClose = () => {
    this.setState({ showModal: false });
  };

  render() {
    const { user, customerDetails, performDeposit } = this.props;
    const { accountNumber, remarks, amount, isLoading, showModal } = this.state;

    const handlePerformDeposit = async () => {
      this.setState({ isLoading: true });
      await performDeposit(user.token, accountNumber, { remarks, amount });
      this.setState({ isLoading: false });
      this.setState({ showModal: false });
    };
    return (
      <main className="container">
        <SideNavPanel user={user} />
        <div className="main-wrapper">
          <TopSection />
          <PageTitle name={'deposit'} />

          <form
            onSubmit={this.handleConfirmDetails}
            className="entry--closer entry"
          >
            <input
              type="number"
              className="entry__input entry__input--bigger"
              placeholder="account number"
              name="accountNumber"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleInputChange}
            />
            <input
              type="number"
              className="entry__input entry-deposit__amount entry__input--bigger"
              placeholder="amount"
              name="amount"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleInputChange}
            />
            <textarea
              className="entry__input entry__input--remark entry__input--bigger"
              placeholder="remark"
              name="remarks"
              required
              onChange={this.handleInputChange}
              onBlur={this.handleInputChange}
            ></textarea>
            <button
              disabled={isLoading}
              className="btn entry__btn entry__input--bigger modal__open"
            >
              deposit
            </button>
          </form>
        </div>

        <div
          onClick={this.handleModalClose}
          className={`modal ${showModal ? 'modal-open' : ''}`}
        >
          {!this.state.loading && Object.keys(customerDetails).length > 0 ? (
            <div href="#" className="transaction-full">
              <div className="transaction__details__title">
                <h1 className="content-title__text modal__title">
                  Confirm Deposit
                </h1>
              </div>
              <div className="transaction-full__main modal__main">
                <div className="transaction-full__type transaction-full__section">
                  <h4 className="transaction-full__title">Account Name</h4>
                  <p className="transaction-full__body">
                    {' '}
                    {`${customerDetails.accountDetails.owner.firstname} ${customerDetails.accountDetails.owner.lastname}`}
                  </p>
                </div>
                <div className="transaction-full__type transaction-full__section">
                  <h4 className="transaction-full__title">Account Number</h4>
                  <p className="transaction-full__body">
                    {customerDetails.accountDetails.accountnumber}
                  </p>
                </div>
                <div className="transaction-full__type transaction-full__section">
                  <h4 className="transaction-full__title">Amount</h4>
                  <p className="transaction-full__body">{amount}</p>
                </div>
                <div className="transaction-amount transaction-full__section">
                  <h4 className="transaction-full__title">remark</h4>
                  <p className="transaction-full__body">{remarks}</p>
                </div>
                <div className="transaction-full__section modal-response">
                  <button
                    className="btn modal__close action-center__btn btn-secondary"
                    onClick={this.handleModalClose}
                  >
                    cancel
                  </button>
                  <button
                    onClick={handlePerformDeposit}
                    className="btn action-center__btn"
                  >
                    confirm
                  </button>
                </div>
              </div>
            </div>
          ) : (
            'loading ...'
          )}
        </div>
      </main>
    );
  }
}

DepositPage.propTypes = {
  user: PropTypes.object,
  fetchAccountDetails: PropTypes.func,
  customerDetails: PropTypes.object,
  performDeposit: PropTypes.func
};

export const mapStateToProps = state => {
  return {
    user: state.users.userAuthDetails,
    customerDetails: state.customerDetails
  };
};

export const mapDispatchToProps = dispatch => ({
  fetchAccountDetails: (token, accountNumber) => {
    return dispatch(fetchAccountDetails(token, accountNumber));
  },
  performDeposit: (token, accountNumber, depositData) => {
    return dispatch(performDeposit(token, accountNumber, depositData));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DepositPage);
