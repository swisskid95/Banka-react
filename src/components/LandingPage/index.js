import React from 'react';
import { Link } from 'react-router-dom';

const LandingPage = () => {
  return (
    <main className="container__entry">
      <div className="landing__page">
        <div className="section-wrapper">
          <section className="main-section">
            <h1 className="logo--big">Banka</h1>
            <h2 className="first-text">Banking made easier</h2>
            <p className="second-text">
              Banka helps you track all transaction on your account(s)
            </p>
            <Link to="/sign-up" className="cta">
              SIGN-UP{' '}
            </Link>
          </section>
          <section className="infograph-section">
            <img
              className="monitor"
              src="../../src/assets/images/infograph.svg"
              alt="infograph"
            />
          </section>
        </div>
      </div>
    </main>
  );
};

export default LandingPage;
