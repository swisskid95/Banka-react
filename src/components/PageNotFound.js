import React from 'react';
import TopSection from '../components/commons/TopSection';

const PageNotFound = () => (
  <main className="container">
    <div className="main-wrapper">
      <TopSection />
      <h1 className="page-not-found">Oops! Page not found.</h1>
    </div>
  </main>
);

export default PageNotFound;
