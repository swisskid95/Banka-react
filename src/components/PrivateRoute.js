/* istanbul ignore file */
import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const PrivateRoute = ({ component: Component, auth, ...rest }) => (
  <React.Fragment>
    <Route
      {...rest}
      render={props =>
        auth.isAuthenticated === true ? (
          <Component {...props} />
        ) : (
          <Redirect to="/login" />
        )
      }
    />
  </React.Fragment>
);

PrivateRoute.propTypes = {
  auth: PropTypes.object.isRequired,
  component: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.users
});

export default connect(mapStateToProps)(PrivateRoute);
