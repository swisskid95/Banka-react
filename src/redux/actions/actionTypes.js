export const LOGIN_USER = 'LOGIN_USER';
export const LOGIN_USER_ERROR = 'LOGIN_USER_ERROR';
export const SIGN_UP_USER = 'SIGN_UP_USER';
export const SIGN_UP_USER_ERROR = 'SIGN_UP_USER_ERROR';
export const ADD_ACCOUNT_DETAILS = 'ADD_ACCOUNT_DETAILS';
export const ADD_ALL_ACCOUNTS = 'ADD_ALL_ACCOUNTS';
export const ADD_ALL_ACCOUNTS_ERROR = 'ADD_ALL_ACCOUNTS_ERROR';
export const SET_CURRENT_USER = 'SET_CURRENT_USER';
export const GET_USER_DETAILS = 'GET_USER_DETAILS';
