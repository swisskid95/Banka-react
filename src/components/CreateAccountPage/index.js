import React, { Component } from 'react';
import SideNavPanel from '../commons/SideNavPanel';
import TopSection from '../commons/TopSection';
import { connect } from 'react-redux';
import { createAccountRequest } from '../../redux/actions/createAccountActions';
import PropTypes from 'prop-types';
export class CreateAccountPage extends Component {
  state = {
    type: '',
    accountPurpose: '',
    isLoading: false
  };

  handleInputChange = event => {
    const { target } = event;
    this.setState({ [target.name]: target.value });
  };

  handleSubmit = async event => {
    event.preventDefault();
    const user = JSON.parse(localStorage.getItem('user'));
    this.setState({ isLoading: true });
    await this.props.createAccountRequest(user.token, this.state);
    this.setState({ isLoading: false });
  };

  render() {
    const { isLoading } = this.state;
    return (
      <main className="container">
        <SideNavPanel />
        <div className="main-wrapper">
          <TopSection />
          <div className="content-title">
            <h1 className="content-title__text">create account</h1>
          </div>
          <form onSubmit={this.handleSubmit} className="entry--closer entry">
            <select
              name="type"
              onBlur={this.handleInputChange}
              className="entry__input--account-type entry__input entry__input--bigger"
            >
              <option value="current">current</option>
              <option value="savings">savings</option>
            </select>
            <input
              type="text"
              onChange={this.handleInputChange}
              className="entry__input entry__input--bigger"
              name="accountPurpose"
              placeholder="account purpose"
            />
            <button
              className={`btn entry__btn entry__input--bigger ${
                isLoading ? 'btn-secondary' : ''
              }`}
            >
              {isLoading ? 'Loading...' : 'create account'}
            </button>
          </form>
        </div>
      </main>
    );
  }
}

CreateAccountPage.propTypes = {
  createAccountRequest: PropTypes.func.isRequired
};

export const mapDispatchToProps = dispatch => {
  return {
    createAccountRequest: async (token, data) =>
      dispatch(await createAccountRequest(token, data))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(CreateAccountPage);
