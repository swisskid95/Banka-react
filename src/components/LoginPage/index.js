/* eslint-disable no-console */
import React, { Component } from 'react';
import EntrySideBar from '../commons/EntrySideBar';
import TopSection from '../commons/TopSection';
import PageTitle from '../commons/PageTitle';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loginUserRequest } from '../../redux/actions/loginActions';

export class LoginPage extends Component {
  state = {
    email: '',
    password: '',
    isLoading: false
  };

  handleInputChange = event => {
    event.preventDefault();
    const { target } = event;

    this.setState({ [target.name]: target.value });
  };

  handleSubmit = async event => {
    event.preventDefault();
    this.setState({ isLoading: true });
    await this.props.loginUserRequest(this.state, this.props.history);
    this.setState({ isLoading: false });
  };

  render() {
    const { isLoading } = this.state;
    return (
      <main className="container__entry">
        <EntrySideBar
          welcomeText="HELLO, Friend"
          mainText="Not yet a member, click on the button below to Join in the benefits."
          buttonText="sign-up"
        />
        <TopSection />
        <PageTitle name={'login'} />
        <form className="entry" onSubmit={this.handleSubmit}>
          <div className="field">
            <input
              className="entry__input"
              type="email"
              placeholder="Email"
              name="email"
              required
              onChange={this.handleInputChange}
            />
            <span className="helper-text"></span>
          </div>
          <div className="field">
            <input
              className="entry__input"
              type="password"
              placeholder="Password"
              name="password"
              required
              onChange={this.handleInputChange}
            />
            <span className="helper-text"></span>
          </div>
          <button
            className={`entry__btn btn ${isLoading ? 'btn-secondary' : ''}`}
          >
            {isLoading ? 'Loading...' : 'Login'}
          </button>
        </form>
        <div className="switch">
          <p>
            Not yet a member?
            <Link to="/sign-up" className="switch__link">
              sign-up
            </Link>
          </p>
        </div>
      </main>
    );
  }
}

LoginPage.propTypes = {
  loginUserRequest: PropTypes.func,
  history: PropTypes.any
};

export const mapDispatchToProps = dispatch => {
  return {
    loginUserRequest: async (loginCredentials, history) =>
      dispatch(await loginUserRequest(loginCredentials, history))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(LoginPage);
