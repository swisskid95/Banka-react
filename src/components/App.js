import React from 'react';
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import PageNotFound from './PageNotFound';
import LandingPage from './LandingPage';
import SignupPage from './SignupPage';
import LoginPage from './LoginPage';
import SummaryPage from './SummaryPage';
import CreateAccountPage from './CreateAccountPage';
import { ToastContainer } from 'react-toastify';
import DepositPage from './DepositPage';
import WithdrawalPage from './WithdrawalPage';
import PrivateRoute from './PrivateRoute';

class App extends React.Component {
  render() {
    return (
      <Router>
        <ToastContainer />
        <Switch>
          <Route exact path="/" component={LandingPage} />
          <Route path="/sign-up" component={SignupPage} />
          <Route path="/login" component={LoginPage} />
          <PrivateRoute path="/summary" component={SummaryPage} />
          <PrivateRoute path="/create-account" component={CreateAccountPage} />
          <PrivateRoute path="/deposit" component={DepositPage} />
          <PrivateRoute path="/withdraw" component={WithdrawalPage} />
          <Route component={PageNotFound} />
        </Switch>
      </Router>
    );
  }
}

export default hot(module)(App);
