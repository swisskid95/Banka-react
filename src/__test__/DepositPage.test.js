/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { DepositPage, mapDispatchToProps } from '../components/DepositPage';
import { GET_USER_DETAILS } from '../redux/actions/actionTypes';
import {
  fetchAccountDetails,
  performDeposit
} from '../redux/actions/cashierStaffAction';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import mockAxios from 'axios';
import configureMockStore from 'redux-mock-store';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const handleSubmit = jest.fn();

const credential = {
  type: 'savings'
};

const user = {
  token:
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MjMsIâ€¦c4OX0.DfJClt28K3qIgJJWp9FMypDO0tKD1NoiNtO9YvEOvsQ'
};
const renderDepositPage = args => {
  const defaultProps = {
    fetchAccountDetails: jest.fn(),
    performDeposit: jest.fn()
  };
  const props = {
    ...defaultProps,
    ...args,
    onSubmit: { handleSubmit },
    user,
    customerDetails: {}
  };
  return mount(
    <BrowserRouter>
      <DepositPage {...props} />
    </BrowserRouter>
  );
};
const renderDepositPage1 = args => {
  const customerDetails = {
    accountnumber: 1212121212,
    accountDetails: { owner: { firstname: 'name', lastname: 'name' } }
  };
  const defaultProps = {
    fetchAccountDetails: jest.fn(),
    performDeposit: jest.fn()
  };
  const props = {
    ...defaultProps,
    ...args,
    onSubmit: { handleSubmit },
    user,
    customerDetails
  };
  return mount(
    <BrowserRouter>
      <DepositPage {...props} />
    </BrowserRouter>
  );
};

describe('without customer details in Components Tests', () => {
  it('renders create account page In template', () => {
    const wrapper = renderDepositPage();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form').length).toBe(1);
  });

  it(`Simulates an onChange event on select account type input`, () => {
    const wrapper = renderDepositPage();
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { currentTarget: { value: 'savings' } });
  });

  it(`Simulates an onchange event on form password input`, () => {
    const wrapper = renderDepositPage();
    wrapper
      .find('input')
      .at(1)
      .simulate('change', { currentTarget: { value: 'Sanusi' } });
  });
  it(`Simulates an onchange event on form password input`, () => {
    const wrapper = renderDepositPage();
    wrapper
      .find('textarea')
      .at(0)
      .simulate('change', { currentTarget: { value: 'Sanusi' } });
  });
  it('Simulates a form submit event', () => {
    const wrapper = renderDepositPage();
    wrapper.find('form').simulate('submit');
  });

  it('should dispatch signup request action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchAccountDetails(user.token, credential);
    mapDispatchToProps(dispatch).performDeposit(user.token, credential);
  });
});

describe('without customer details in Components Tests', () => {
  it('renders create account page In template', () => {
    const wrapper = renderDepositPage1();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form').length).toBe(1);
  });

  it(`Simulates an onChange event on select account type input`, () => {
    const wrapper = renderDepositPage1();
    wrapper
      .find('input')
      .at(0)
      .simulate('change', { currentTarget: { value: 'savings' } });
  });

  it(`Simulates an onchange event on form password input`, () => {
    const wrapper = renderDepositPage1();
    wrapper
      .find('input')
      .at(1)
      .simulate('change', { currentTarget: { value: 'Sanusi' } });
  });

  it(`Simulates an onchange event on form password input`, () => {
    const wrapper = renderDepositPage1();
    wrapper
      .find('textarea')
      .at(0)
      .simulate('change', { currentTarget: { value: 'Sanusi' } });
  });
  it('Simulates a form submit event', () => {
    const wrapper = renderDepositPage1();
    wrapper.find('form').simulate('submit');
  });

  it('should dispatch signup request action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchAccountDetails(user.token, credential);
  });
});

describe('create account request Async action Tests', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
    jest.resetAllMocks();
  });
  afterEach(() => {
    store.clearActions();
  });
  it('Should Trigger the ADD_ACCOUNT_DETAILS dispatch function', async () => {
    const mockData = {
      data: {
        data: {
          accountNumber: 9093234532,
          type: 'savings'
        }
      }
    };
    mockAxios.get.mockResolvedValue({
      status: 200,
      data: mockData
    });

    const expectedActions = [
      { type: GET_USER_DETAILS, payload: mockData.data }
    ];
    await store.dispatch(fetchAccountDetails(credential, user.token));
    expect(store.getActions()).toEqual(expectedActions);
  });
  it('Should Trigger the ADD_ACCOUNT_DETAILS dispatch function', async () => {
    const mockData = {
      data: {
        data: {
          accountNumber: 9093234532,
          type: 'savings'
        }
      }
    };
    mockAxios.post.mockResolvedValue({
      status: 201,
      data: mockData
    });

    await store.dispatch(performDeposit(credential, user.token));
    expect(store.getActions()).toEqual([]);
  });
});
