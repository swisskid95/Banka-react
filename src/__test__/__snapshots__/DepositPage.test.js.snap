// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`without customer details in Components Tests renders create account page In template 1`] = `
<main
  className="container"
>
  <aside
    className="sidebar sidebar--light"
  >
    <span
      className="sidebar__close"
    >
      ×
    </span>
    <div
      className="avatar"
    >
      <img
        alt="user"
        className="avatar__img"
        src="https://res.cloudinary.com/swisskid95/image/upload/v1566613597/Banka-assets/defaultAvatar_roxuye.png"
      />
    </div>
    <h3
      className="user-name"
    >
      undefined undefined
    </h3>
    <ul
      className="nav-container"
    >
      <li>
        <a
          aria-current={null}
          className="nav"
          href="/summary"
          onClick={[Function]}
        >
          summary
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="nav"
          href="/create-account"
          onClick={[Function]}
        >
          new account
        </a>
      </li>
      <li>
        <a
          className="nav  logout"
          href="/login"
          onClick={[Function]}
        >
          sign-out
        </a>
      </li>
    </ul>
  </aside>
  <div
    className="main-wrapper"
  >
    <div
      className="top-section"
    >
      <img
        alt="Sidebar open button"
        className="sidebar__open"
        src="https://res.cloudinary.com/swisskid95/image/upload/v1566575800/Banka-assets/menu_ldfqrk.svg"
      />
      <a
        className="logo"
        href="/summary"
        onClick={[Function]}
      >
        <h1
          className="logo"
        >
          Banka
        </h1>
      </a>
    </div>
    <div
      className="content-title"
    >
      <h1
        className="content-title__text"
      >
        deposit
      </h1>
    </div>
    <form
      className="entry--closer entry"
      onSubmit={[Function]}
    >
      <input
        className="entry__input entry__input--bigger"
        name="accountNumber"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="account number"
        required={true}
        type="number"
      />
      <input
        className="entry__input entry-deposit__amount entry__input--bigger"
        name="amount"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="amount"
        required={true}
        type="number"
      />
      <textarea
        className="entry__input entry__input--remark entry__input--bigger"
        name="remarks"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="remark"
        required={true}
      />
      <button
        className="btn entry__btn entry__input--bigger modal__open"
      >
        deposit
      </button>
    </form>
  </div>
  <div
    className="modal "
    onClick={[Function]}
  >
    loading ...
  </div>
</main>
`;

exports[`without customer details in Components Tests renders create account page In template 2`] = `
<main
  className="container"
>
  <aside
    className="sidebar sidebar--light"
  >
    <span
      className="sidebar__close"
    >
      ×
    </span>
    <div
      className="avatar"
    >
      <img
        alt="user"
        className="avatar__img"
        src="https://res.cloudinary.com/swisskid95/image/upload/v1566613597/Banka-assets/defaultAvatar_roxuye.png"
      />
    </div>
    <h3
      className="user-name"
    >
      undefined undefined
    </h3>
    <ul
      className="nav-container"
    >
      <li>
        <a
          aria-current={null}
          className="nav"
          href="/summary"
          onClick={[Function]}
        >
          summary
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="nav"
          href="/create-account"
          onClick={[Function]}
        >
          new account
        </a>
      </li>
      <li>
        <a
          className="nav  logout"
          href="/login"
          onClick={[Function]}
        >
          sign-out
        </a>
      </li>
    </ul>
  </aside>
  <div
    className="main-wrapper"
  >
    <div
      className="top-section"
    >
      <img
        alt="Sidebar open button"
        className="sidebar__open"
        src="https://res.cloudinary.com/swisskid95/image/upload/v1566575800/Banka-assets/menu_ldfqrk.svg"
      />
      <a
        className="logo"
        href="/summary"
        onClick={[Function]}
      >
        <h1
          className="logo"
        >
          Banka
        </h1>
      </a>
    </div>
    <div
      className="content-title"
    >
      <h1
        className="content-title__text"
      >
        deposit
      </h1>
    </div>
    <form
      className="entry--closer entry"
      onSubmit={[Function]}
    >
      <input
        className="entry__input entry__input--bigger"
        name="accountNumber"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="account number"
        required={true}
        type="number"
      />
      <input
        className="entry__input entry-deposit__amount entry__input--bigger"
        name="amount"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="amount"
        required={true}
        type="number"
      />
      <textarea
        className="entry__input entry__input--remark entry__input--bigger"
        name="remarks"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="remark"
        required={true}
      />
      <button
        className="btn entry__btn entry__input--bigger modal__open"
      >
        deposit
      </button>
    </form>
  </div>
  <div
    className="modal "
    onClick={[Function]}
  >
    <div
      className="transaction-full"
      href="#"
    >
      <div
        className="transaction__details__title"
      >
        <h1
          className="content-title__text modal__title"
        >
          Confirm Deposit
        </h1>
      </div>
      <div
        className="transaction-full__main modal__main"
      >
        <div
          className="transaction-full__type transaction-full__section"
        >
          <h4
            className="transaction-full__title"
          >
            Account Name
          </h4>
          <p
            className="transaction-full__body"
          >
             
            name name
          </p>
        </div>
        <div
          className="transaction-full__type transaction-full__section"
        >
          <h4
            className="transaction-full__title"
          >
            Account Number
          </h4>
          <p
            className="transaction-full__body"
          />
        </div>
        <div
          className="transaction-full__type transaction-full__section"
        >
          <h4
            className="transaction-full__title"
          >
            Amount
          </h4>
          <p
            className="transaction-full__body"
          />
        </div>
        <div
          className="transaction-amount transaction-full__section"
        >
          <h4
            className="transaction-full__title"
          >
            remark
          </h4>
          <p
            className="transaction-full__body"
          />
        </div>
        <div
          className="transaction-full__section modal-response"
        >
          <button
            className="btn modal__close action-center__btn btn-secondary"
            onClick={[Function]}
          >
            cancel
          </button>
          <button
            className="btn action-center__btn"
            onClick={[Function]}
          >
            confirm
          </button>
        </div>
      </div>
    </div>
  </div>
</main>
`;
